source/core/NumericString.cpp

```diff
@@ -2408,14 +2408,65 @@ VIREO_FUNCTION_SIGNATURE4(ArraySpreadsheet, StringRef, StringRef, StringRef, Typ
     }
     return _NextInstruction();
 }
+
+// Find the next delimiter in a spreadsheet string starting at offset and return the matching offset (or -1)
+// and the offset past the match; delimiterParam can be either a single StringRef or an array of StringRefs
+static inline IntIndex FindDelimiter(SubString *input, TypedArrayCoreRef delimiterParam, IntIndex offset, IntIndex *next, Boolean isDelimArray) {
+    if (!isDelimArray) {
+        StringRef delimiterString = (StringRef)delimiterParam;
+        SubString delimiter = delimiterString->MakeSubStringAlias();
+
+        IntIndex index = input->FindFirstMatch(&delimiter, offset, false);
+        *next = index >= 0 ? index + delimiter.Length() : -1;
+        return index;
+    } else {
+        StringRefArray1D *delimArray = (StringRefArray1D*)delimiterParam;
+        Int32 index = -1, pastMatch = -1;
+        Int32 length = delimArray->Length();
+        for (IntIndex i = 0; i < length; ++i) {
+            StringRef delimiterString = delimArray->At(i);
+            SubString delimiter = delimiterString->MakeSubStringAlias();
+            IntIndex match = input->FindFirstMatch(&delimiter, offset, false);
+            // Find the first match in the delimiter array, if multiple delims match, take the longest
+            if (match >= 0 && (index == -1 || match < index
+                || (match == index && index + delimiter.Length() > pastMatch))) {
+                index = match;
+                pastMatch = index + delimiter.Length();
+            } else if (input->Length() - offset < delimiter.Length()) {
+                // This dutifully replicates what is arguably a CG bug; if we have a successful partial match
+                // of one of the delimiter string when we 'run out' of input string, we ignore the delimiter.
+                delimiter.AliasAssignLen(delimiter.Begin(), input->Length() - offset);
+                match = input->FindFirstMatch(&delimiter, offset, false);
+                if (match >= 0) {
+                    index = pastMatch = -1;
+                    break;
+                }
+            }
+        }
+        *next = pastMatch;
+        return index;
+    }
+}
+
+// Wrapper around FormatScan:  %s for Spreadsheet Strings does not stop at whitespace the way it does
+// for the other format functions.
+static inline void FormatScanSS(SubString *elemString, SubString *format,  StaticTypeAndData *Value, Boolean formatIsString) {
```

Comment:
Consider renaming to FormatScanForSpreadSheet. When looking at SS I was wondering about the meaning, until I figured out, but it is an unnecessary distraction when reading code.

Reply:
Done.